diff --git a/gir/meson.build b/gir/meson.build
index e5a7a131..fbc3cc8b 100644
--- a/gir/meson.build
+++ b/gir/meson.build
@@ -10,9 +10,12 @@ gir_files = [
   {'name': 'xlib', 'file': 'xlib-2.0.gir' },
   {'name': 'xfixes', 'file': 'xfixes-4.0.gir' },
   {'name': 'xrandr', 'file': 'xrandr-1.3.gir' },
-  {'name': 'win32', 'file': 'win32-1.0.gir' },
 ]
 
+if host_system == 'windows'
+  gir_files += {'name': 'win32', 'file': 'win32-1.0.gir' }
+endif
+
 uninstalled_gir_files = []
 
 # Copy gir files to build directory to have them all in a single place.
